### PR TITLE
DOMA: clear VALPOS

### PR DESCRIPTION
**Pull request by ibnbr**
issue #1053

### CODE REVIEW

**Comment by larshp** [comment]
closes #1053

**Comment by larshp** [comment]
please open PR with changed files to https://github.com/abapGit-tests/DOMA

**Comment by ibnbr** [comment]
this object has no value range, so I have no changes to open a PR. can I create de value range and open a PR?

**Comment by larshp** [comment]
https://github.com/abapGit-tests/DOMA_fixed_values then

**Comment by ibnbr** [comment]
PR Done.
I had to make some adjustments.

**Comment by larshp** [comment]
Suggest also clearing DD07V-DOMNAME, as it is always the same as the object name

**Comment by ibnbr** [comment]
I made some changes, and I'm doing some tests, when DOMA has translation is not working properly. I don't know if it was caused by my changes, I'll investigate.

**Comment by ibnbr** [comment]
Removed VALPOS and DOMNAME. All tests OK now.

File: src/objects/zcl_abapgit_object_doma.clas.abap [hunk 1]
```diff
@@ -82,10 +83,13 @@ CLASS ZCL_ABAPGIT_OBJECT_DOMA IMPLEMENTATION.
       " Domain values
       lt_dd07v_tmp = it_dd07v.
       LOOP AT lt_dd07v_tmp ASSIGNING <ls_dd07v>.
+        lv_valpos = <ls_dd07v>-valpos.
         READ TABLE lt_dd07_texts ASSIGNING <ls_dd07_text>
-          WITH KEY ddlanguage = <lv_lang> valpos = <ls_dd07v>-valpos.
+          WITH KEY ddlanguage = <lv_lang> domvalue_l = <ls_dd07v>-domvalue_l
+                   domvalue_h = <ls_dd07v>-domvalue_h.
```
Comment (larshp) [inline src/objects/zcl_abapgit_object_doma.clas.abap]: is this change related to clearing of the fields?

Comment (ibnbr) [reply src/objects/zcl_abapgit_object_doma.clas.abap]: Yes, because "it_dd07_texts" don't have valpos filled. I'm not sure if we can use "sy-tabix" to read, probably it'll work but I think reading using the key is most reliable

Comment (larshp) [reply src/objects/zcl_abapgit_object_doma.clas.abap]: hmm

**Comment by larshp** [comment]
If I pull a DOMA serialized with the new code, with an old version of abapGit, what will happen?

**Comment by ibnbr** [comment]
I tested only old DOMA serialized on a new version of abapGit and it's work.

I'll prepare the system to test this, but I think the function module  that update DOMA will return a exception.

**Comment by ibnbr** [comment]
As I thought an exception is raised in the function module, but this message doesn't appear in the UI.

image

I'll try understand the UI of abapGit, because I'm lost on this.

**Comment by larshp** [comment]
is the error caused by clearing VALPOS or DOMNAME? perhaps we can clear just one of them and have it working on old versions

**Comment by ibnbr** [comment]
For both.  DDIF_DOMA_PUT returns 'NAME_INCONSISTENT' for DOMNAME empty and 'DOMA_INCONSISTENT' for VALPOS empty.

**Comment by larshp** [comment]
sorry about the delay

Suggest this approach: add the code to fill the empty values now,

In 4(?) months: add the code that clears the fields

This way users will have a chance to update to latest abapGit, and we can avoid some problems

**Comment by ibnbr** [comment]
OK. So I will split this changes in 2 PRs, right?

**Comment by larshp** [comment]
yes please

**Comment by larshp** [comment]
closing
